### PR TITLE
Handle environment secrets concurrently

### PR DESCRIPTION
**Pull request by travisgroth**
Related to #782 - Process the environment secrets in parallel.

### CODE REVIEW

**Comment by travisgroth** [comment]
@mumoshu I'm a bit lost on the best way to get pass workerLimit/Concurrency all the way down the stack to where we load the secrets.  I'd welcome a suggestion on how to handle that; right now, it's hard coded to 0.  I don't think this is problematic in practice, but obviously we should respect the limits everywhere.

Otherwise, I think this is ready.

**Comment by mumoshu** [comment]
@travisgroth Thanks! Regarding the hard-coded concurrency - That sounds ok.

The purpose of --concurrency/worker-limit for me is to limit how much cluster resources are eventually consumed by running Helmfile. Decrypting env secrets doesn't affect the cluster so I have no specific reason to make concurrency configurable here.

**Comment by travisgroth** [comment]
👍

File: pkg/state/create.go [hunk 1]
```diff
@@ -201,59 +202,100 @@ func (st *HelmState) loadEnvValues(name string, ctxEnv *environment.Environment,
...
+			close(secrets)
+		},
+		func(id int) {
+			for path := range secrets {
 				release := &ReleaseSpec{}
 				flags := st.appendConnectionFlags([]string{}, release)
 				decFile, err := helm.DecryptSecret(st.createHelmContext(release, 0), path, flags...)
 				if err != nil {
-					return nil, err
+					results <- secretResult{nil, err, path}
```
Comment (mumoshu) [inline pkg/state/create.go]: I think we need `return` after L256, L260, L264 and L272 to not submit redundant results in L274.

Comment (travisgroth) [reply pkg/state/create.go]: Good catch.  With the way this was structured, it should be a continue to catch all the errors in a given run.  I think that also keeps it from blocking when gathering the results, so we have some sort of result from every task.  It should also reduce test/error cycles when using helmfile.

Pushed an update for that, and fixing an extra ref for the envVals map.

Comment (mumoshu) [reply pkg/state/create.go]: Ah that makes sense! `so we have some sort of result from every task` was the critical part of it I had forgot at the time of writing the comment. LGTM.

**Comment by travisgroth** [comment]
👍

**Pull request merged**